### DISCUSSION TITLE
All Streaming Functions Suddenly Timing Out at 15 sec

### DISCUSSION THREAD

**Discussion by mrjasonroy** [vercel/community#4444] [Help] [answered]
### Summary

We have a chatbot built on top of the vercel/ai library, and as of today, any response that is over 15 seconds long cuts off. It seems like a change on the vercel side as there were no code changes and it's been working for months. 

The api route is in the app directory, has force-dynamic on it, etc. This is happening to multiple sites and it wasn't happening before. We are on a pro plan.

### Example

https://chat-demo.biocompare.com/

### Steps to Reproduce

Use the chatbot ai library and produce a response, deploy to vercel.

Note, this is NOT happening on local or non vercel deployments. 

You can visit the above, register/login, use the built in product and select a suggested prompt. It seems like there was a change at vercel

**Comment by mrjasonroy** [comment]
Note, this is happening in a monorepo and there was never a vercel.json so I am not sure why this is suddenly happening. If that is the answer, should that be included in the documentation for the ai library?

**Comment by amyegan** [comment, marked as answer]
The current default [`maxDuration`](https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration) for serverless functions is 15s for Pro with the option to configure it for up to 300s max. If you specify a higher `maxDuration` for your function, does it still get cut off at 15 seconds?

https://vercel.com/docs/functions/serverless-functions/runtimes#maxduration

**Comment by mrjasonroy** [comment]
That did it! I think we didn't notice it before because the default was downgraded from 60 to 15. Hopefully this help ticket helps if others suddenly see it.

**Reply by amyegan** [reply]
I'm glad that worked for you!
